my_pyscf/dmet/_dmet.py

```diff
@@ -0,0 +1,577 @@
+import numpy as np
+import scipy
+from pyscf import gto, ao2mo, lib, scf
+from mrh.my_pyscf.dmet.localization import Localization
+from mrh.my_pyscf.dmet.fragmentation import Fragmentation
+from mrh.my_pyscf.dmet.basistransformation import BasisTransform
+
+# Author: Bhavnesh Jangid <jangidbhavnesh@uchicago.edu>
+
+# This class will give you the mean-field object, which can be used to run the MR and post-HF
+# calculations.
+
+get_basis_transform = BasisTransform._get_basis_transformed
+
+def is_close_to_integer(num, tolerance=1e-6):
+    warning_msg = f'SVD for this RDM has some problem {(np.abs(num - np.round(num)) < tolerance)}'
+    assert (np.abs(num - np.round(num)) < tolerance), warning_msg 
+
+
+def perform_schmidt_decomposition_type1(rdm, nfragorb, nlo, bath_tol=1e-5):
+    '''
+    This way of schmidt decomposition corresponds to previous implementation
+    where one will consider all the singly occupied orbital in the bath, whether they
+    are in impurity space or not.
+
+    Not a wise way of doing it, still having it to make sure we have functionalies of previous
+    implementations.
+
+    Args:
+        rdm: np.ndarray (2, n, n); 
+             n=number of enviroment orbitals
+        nfragorb: int
+            no of fragment orb
+        nlo: int
+            no of localized orb, basically this will be same as that of total number of atomic orbital.
+        bath_tol: float
+            threshold to keep an orbital in the embedding space or through it out.
+    return
+        u_selected: np.array (nlo, neo)
+            unitary matrix which will transform the localized orbitals to the bath orbitals 
+        u_core: np.array(nlo, ncore)
+            unitary matrix which will transform the localized orbitals to the core orbitals 
+    '''
+    alpha_evecs, alpha_evals = np.linalg.svd(rdm[0])[:2]
+    beta_evecs, beta_evals= np.linalg.svd(rdm[1])[:2]
+    occ, u, alpha = (alpha_evals, alpha_evecs, 1) \
+        if np.sum(alpha_evals) >= np.sum(beta_evals) else (beta_evals, beta_evecs, 0)
+
+    # Select all the orbitals which are above the bath_tolerance
+    # in this way, if their are any singly occupied orbital in environment
+    # that will taken into the bath orbital.
+    sinorbthresh = 0.9
+    occ1, occ2 = occ, beta_evals if alpha else alpha_evals
+    bathocc = np.array([(ai - bi) if ai > sinorbthresh and bi > sinorbthresh \
```

Comment:
So if I read this right, if an environment orbital has an alpha occupancy of 0.92 and a beta occupancy of 0.91, it's a bath orbital, but if the alpha and beta occupancy are both exactly 0.91, then it's a core orbital. Is that right? Is that the intended behavior? Is that how the previous implementation worked?

I think the more elegant thing is just to diagonalize the core 1-RDM after line 33, whose eigenvalues at that point should all be 2, 1, or 0, and move any columns corresponding to eigenvalue = 1 from `u_core` to `u_selected`. Then again, I do remember emphasizing to you that we wanted to reproduce the previous implementation's behavior, so if that's what you're doing here, then I understand.

Reply:
It's fine the way it is now.

```
It is not necessary that the eigenvalues of the environment 1-RDM (in the orthogonalized localized orbital basis) is equal to 2,1, or 0. There can be fractional eigenvalues, which indicate the entanglement with the impurity orbitals.
```

But if you first do the SVD and generate the embedding basis, then the eigenvalues of the core-orbital 1-RDM must be integers after that point.